cltk/tokenize/latin/word.py

```diff
@@ -0,0 +1,105 @@
+""" Code for word tokenization: Latin
+"""
+
+__author__ = ['Patrick J. Burns <patrick@diyclassics.org>']
+__license__ = 'MIT License.'
+
+import re
+
+from nltk.tokenize.treebank import TreebankWordTokenizer
+
+from cltk.tokenize.word import BasePunktWordTokenizer
+from cltk.tokenize.latin.params import ABBREVIATIONS, latin_exceptions, latin_replacements
+from cltk.tokenize.latin.sentence import SentenceTokenizer
+
+def WordTokenizer():
+    return LatinPunktWordTokenizer()
+
+class LatinPunktWordTokenizer(BasePunktWordTokenizer):
+    """
+    PunktSentenceTokenizer trained on Latin
+    """
+    def __init__(self: object, language:str = 'latin', sent_tokenizer=SentenceTokenizer):
+        """
+        :param language : language for word tokenization
+        :type language: str
+        """
+        super().__init__(language='latin')
+        self.sent_tokenizer = sent_tokenizer()
+        self._latin_replacements = latin_replacements
+
+    def tokenize(self, text: str, split_enclitics:list = ['ne', 'n', 'que', 've', 'ue', 'st'],
+                                  split_words:list = []):
+        """
+        :rtype: list
+        :param text: text to be tokenized into sentences
+        :type text: str
+        :param model: tokenizer object to used # Should be in init?
+        :type model: object
+        """
+        if self._latin_replacements:
+            split_words = self._latin_replacements
+
+        if split_words:
+            text = self._replace_patterns(text, split_words)
+        sents = self.sent_tokenizer.tokenize(text)
+        if split_enclitics:
+            sents = self._split_enclitics(sents, split_enclitics)
+        tokenizer = TreebankWordTokenizer()
+        return [item for sublist in tokenizer.tokenize_sents(sents) for item in sublist]
+
+    def _split_enclitics(self:object, sents:list, enclitics: list):
+        import string
+        exclude_flag = '~'
+        if 'ne' in enclitics and 'n' in enclitics:
+            ne_compile = re.compile(r'^\b(\w+?)([n]e?)[%s]?\b'%re.escape(string.punctuation))
```

Comment:
A good optimization might be to re.compile at init and store the compilations in the self object

Reply:
Great idea—and a lesson I feel like I should have learned by now!